core/src/value/deserialize.rs

```diff
@@ -291,9 +305,140 @@ pub fn parse_datetime(buf: &[u8], dt_utc_offset: FixedOffset) -> Result<DateTime
         .context(InvalidDateTimeZone)
 }
 
+macro_rules! check_range {
+    ($s: expr, $e: expr) => {
+        if $s == $e {
+            RangeIsZero {
```

Comment:
According to [PS3.4 C.2.2.2.5](http://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_C.2.2.2.5.html), equal ends are OK, and the match is inclusive. For instance, a range such as `20210730-20210730` would simply match one date.

The error variant `RangeIsZero` should then be removed, and the tests be updated accordingly.